### PR TITLE
chore(cwc-v2): various updates from `main` into `feat/cwc-v2`

### PR DESCRIPTION
**Pull request by kennylam**
### Description

Adds various commits from `main`.

### Changelog

**New**

- 305c406
- b8e919d
- 2e8caca
- 305c406
- 753d090
- a71b9f9

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

File: packages/carbon-web-components/tests/snapshots/file-uploader.md [hunk 1]
```diff
@@ -269,11 +269,14 @@
 
 ```
 
+<<<<<<< HEAD
 ## `cds-file-uploader-shell`
 
+=======
+>>>>>>> 753d090a6d (chore(deps): upgrade Node 18 (#10239))
```
Comment (emyarod) [inline packages/carbon-web-components/tests/snapshots/file-uploader.md:276]: leftover from merge conflict resolution?

**Review by emyarod — COMMENTED** [review]
saw something in one of the snapshots

**Review by emyarod — APPROVED** [review]
looks good to me!

**Pull request merged**